sos/report/plugins/openstack_designate.py

```diff
@@ -19,6 +19,25 @@ class OpenStackDesignate(Plugin):
     var_puppet_gen = "/var/lib/config-data/puppet-generated/designate"
 
     def setup(self):
+        # collect current pool config
+        create_pools_cmd = "designate-manage pool generate_file --file "
+        collect_pools_cmd = "cat "
+        self.pools_path = self.var_puppet_gen + "/tmp/sos.pools.yaml"
+
+        if self.container_exists(".*designate_central"):
+            self.pools_path = "/tmp/sos.pools.yaml"
+            create_pools_cmd = self.fmt_container_cmd("designate_central",
+                                                      create_pools_cmd)
+            collect_pools_cmd = self.fmt_container_cmd("designate_central",
+                                                       collect_pools_cmd)
+
+        self.exec_cmd(create_pools_cmd + self.pools_path)
+
+        self.add_cmd_output(
+            collect_pools_cmd + self.pools_path,
+            suggest_filename="openstack_designate_current_pools.yaml"
+        )
```

Comment:
If I understood this right, you generate a file here that is cat-ed (and this is collected). Can't that be done directly like https://github.com/sosreport/sos/blob/master/sos/report/plugins/subscription_manager.py#L40-L42 ? I.e. something like:

    ..
    self.add_cmd_output(create_pools_cmd + os.path.join(self.get_cmd_output_path(), openstack_designate_current_pools.yaml),
                        suggest_filename="openstack_designate_current_pools.yaml"
    )

?

Reply:
Yep you understood correctly, I had a quick look for a simpler way but didn't find anything. I'll give your suggestion a go, thanks.

Reply:
I think that method will only work in the non-containerized case since the command within the container won't usually have access to the output directory.

Reply:
This might be a good time for us to explore adding a way to leverage a `ContainerRuntime` to copy files out of a container.

Reply:
@TurboTurtle maybe an extra param for copy spec with the container name? I'm not sure I'm in a position to implement but I can try.

Reply:
I filed https://github.com/sosreport/sos/issues/2439 to track a new feature for copying files from a container, though it might not be worth the effort. It's only needed when the desired file isn't within a mounted directory, which I guess isn't that common since it's not come up before, and the workaround of catting the file is probably good enough for non binary cases.